### PR TITLE
Include rudimentary API docs for all modules

### PR DESCRIPTION
**Pull request by pkausw**
As discussed in our last meeting, we need to include all modules in the API docs.

This PR does that. Note that not all modules have meaningful doc strings or type hints yet.
Nevertheless, the updated version should give at least an overview of the modules and their member function/variables, which would be beneficial for all users.

### CODE REVIEW

**Comment by pkausw** [comment]
Closes #325
Closes #326 
Closes #327

**Review by mafrahm — CHANGES_REQUESTED** [review]
I think the `index.rst` for the plotting modules is still missing. Otherwise, this looks good to me.

**Comment by pkausw** [comment]
> I think the `index.rst` for the plotting modules is still missing. Otherwise, this looks good to me.

The file is uploaded here: https://github.com/columnflow/columnflow/blob/features/api_docs/docs/api/plotting/index.rst (I think it doesn't show as part of the PR because it may have existed before?)

File: columnflow/tasks/framework/mixins.py [hunk 1]
```diff
@@ -10,13 +10,12 @@
...
 import itertools
 from collections import Counter
-from typing import Iterable
 
 import luigi
 import law
 import order as od
 
-from columnflow.types import Sequence, Any
+from columnflow.types import Sequence, Any, Dict, Iterable, Tuple, Union, List
```
Comment (mafrahm) [inline columnflow/tasks/framework/mixins.py:18]: Why are we using these classes (`Dict`, `Tuple`, `List`) for typing instead of the default python types

**Review by mafrahm — APPROVED** [review]
In that case, I agree that we can merge this. Just left one more inline comment for clarification, but no changes are necessary.

File: columnflow/tasks/framework/mixins.py (hunk 1 above)
Comment (pkausw) [reply columnflow/tasks/framework/mixins.py:18]: I think there was an issue with the typing not too long ago, which prompted us to centralize the type definitions in cf (see PR #305 ). I simply follow this approach here, though this needs to be followed-up on for other modules as far as I'm aware.

**Pull request merged**